### PR TITLE
Powershell changes for following changes

### PR DESCRIPTION
**Pull request by SiddharthChatrolaMs**
- taking API version to 2015-03-01
- take changes related to ResourceBase, ResourceBaseExtended and ResourceIdentity
- using redisConfiguration instead of maxMemoryPolicy
- added support for scaling in Set-AzureRedisCache operation

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@SiddharthChatrolaMs__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

**Comment by stankovski** [comment]
LGTM

File: ChangeLog.txt [hunk 1]
```diff
@@ -2,6 +2,7 @@
 * Added Batch cmdlets
     * Start-AzureBatchPoolResize
     * Stop-AzureBatchPoolResize
+* Added support for scaling in Set-AzureRedisCache, using redisConfiguration instead of maxMemoryPolicy, changes related to ResourceBase, ResourceBaseExtended and ResourceIdentity #513
```
Comment (markcowl) [inline ChangeLog.txt]: Normally we do this like this:

* RedisCache cmdlets
    * Set-AzureRedisCache - added support for ...

Comment (SiddharthChatrolaMs) [inline ChangeLog.txt]: Fixed

**Pull request merged**